arangod/Replication2/Streams/LogMultiplexer.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+#include <utility>
+
+#include <Futures/Future.h>
+
+#include <Replication2/ReplicatedLog/ILogParticipant.h>
+#include <Replication2/ReplicatedLog/LogCommon.h>
+#include <Replication2/ReplicatedLog/types.h>
+
+#include <Replication2/Streams/StreamSpecification.h>
+#include <Replication2/Streams/Streams.h>
+
+namespace arangodb::replication2::replicated_log {
+class LogFollower;
+class LogLeader;
+}  // namespace arangodb::replication2::replicated_log
+
+namespace arangodb::replication2::streams {
+
+/**
+ * Common stream dispatcher class for Multiplexer and Demultiplexer. You can
+ * obtain a stream given its id using getStreamById. Alternatively, you can
+ * static_cast the a pointer to StreamBase<Descriptor> for the given stream.
+ * @tparam Self
+ * @tparam Spec
+ * @tparam StreamType
+ */
+template <typename Self, typename Spec, template <typename> typename StreamType>
+struct LogMultiplexerStreamDispatcher : std::enable_shared_from_this<Self>,
+                                        StreamDispatcherBase<Spec, StreamType> {
```

Comment:
For the CRTP parameter I usually prefer the name `Derived`, because IMHO it communicates more clearly how this class needs to be used:
```suggestion
template <typename Derived, typename Spec, template <typename> typename StreamType>
struct LogMultiplexerStreamDispatcher : std::enable_shared_from_this<Derived>,
                                        StreamDispatcherBase<Spec, StreamType> {
```